jkube-kit/build/api/src/main/java/org/eclipse/jkube/kit/build/api/auth/AuthConfig.java

```diff
@@ -35,34 +36,45 @@
 @EqualsAndHashCode
 public class AuthConfig {
 
-  public static final AuthConfig EMPTY_AUTH_CONFIG = new AuthConfig("", "", "", "");
+  public static final AuthConfig EMPTY_AUTH_CONFIG = new AuthConfig("", "", "", "", "");
 
   private final String username;
   private final String password;
   private final String email;
   private final String auth;
+  private final String identityToken;
 
   private String authEncoded;
 
   @Builder
-  public AuthConfig(String username, String password, String email, String auth) {
+  public AuthConfig(String username, String password, String email, String auth, String identityToken) {
     this.username = username;
     this.password = password;
     this.email = email;
     this.auth = auth;
+    this.identityToken = identityToken;
     authEncoded = createAuthEncoded();
   }
 
+  public AuthConfig(String username, String password, String email, String auth) {
+    this(username, password, email, auth, null);
+  }
+
   public String toHeaderValue() {
     return authEncoded;
   }
 
   private String createAuthEncoded() {
     JsonObject ret = new JsonObject();
-    putNonNull(ret, "username", username);
-    putNonNull(ret, "password", password);
-    putNonNull(ret, "email", email);
-    putNonNull(ret, "auth", auth);
+    if(identityToken != null) {
+      putNonNull(ret, "identityToken", identityToken);
```

Comment:
A warning in case of any of the other fields (probably just username) is not blank, should be very helpful.

Reply:
Do you want me to introduce KItLogger to this class?

Reply:
Not really.
Where is the AuthEncoded consumed? An option is to move this logic to the `toHeaderValue()`, the logger should be available for any consumer of this method.